app/coffee/RedisManager.coffee

```diff
@@ -24,16 +31,21 @@ module.exports = RedisManager =
 		logger.log project_id:project_id, doc_id:doc_id, version: version, hash:docHash, "putting doc in redis"
 		ranges = RedisManager._serializeRanges(ranges)
 		multi = rclient.multi()
-		multi.set keys.docLines(doc_id:doc_id), docLines
+		multi.eval setScript, 1, keys.docLines(doc_id:doc_id), docLines
 		multi.set keys.projectKey({doc_id:doc_id}), project_id
 		multi.set keys.docVersion(doc_id:doc_id), version
 		multi.set keys.docHash(doc_id:doc_id), docHash
 		if ranges?
 			multi.set keys.ranges(doc_id:doc_id), ranges
 		else
 			multi.del keys.ranges(doc_id:doc_id)
-		multi.exec (error) ->
+		multi.exec (error, result) ->
 			return callback(error) if error?
+			# check the hash computed on the redis server
+			writeHash = result?[0]
+			if writeHash? and writeHash isnt docHash
+				logger.error project_id: project_id, doc_id: doc_id, writeHash: writeHash, origHash: docHash, "hash mismatch on putDocInMemory"
```

Comment:
Will these error lines trigger a sentry error, or are we just going to monitor the logs?